thoth-app/src/component/contributions_form.rs

```diff
@@ -405,11 +416,26 @@ impl Component for ContributionsFormComponent {
                 self.show_results = value;
                 true
             }
-            Msg::SearchContributor(value) => {
+            Msg::SearchQueryChanged(value) => {
+                self.search_query = value;
+                // cancel previous timeout
+                self.debounce_timeout = self.debounce_timeout.take().and_then(|timeout| {
+                    timeout.cancel();
+                    None
+                });
+                // start new timeout
+                let search_callback = self.search_callback.clone();
+                let timeout = Timeout::new(DEFAULT_DEBOUNCING_TIMEOUT, move || {
+                    search_callback.emit(());
+                });
+                self.debounce_timeout = Some(timeout);
+                false
+            }
+            Msg::SearchContributor => {
                 let body = ContributorsRequestBody {
                     variables: Variables {
-                        filter: Some(value),
-                        limit: Some(9999),
+                        filter: Some(self.search_query.clone()),
+                        limit: Some(25),
```

Comment:
Might be better as a constant.

Note also that when we first load a work, we're not passing a limit in any of the relation queries e.g. `ContributorsQuery`, so we're still fetching the whole list. This is another inefficiency we could trim, as well as leading to inconsistent display: if the user enters text in the search box and then clears it, they won't get the same set of results as when the search box was originally empty.

Reply:
I didn't want to make it a constant as we might want to have different values depending on the component (although I think I used the same in all of them...).

I did notice the inconsistency. But to be fair, we probably should just get rid of the inital value altogether, since the user will be searching regardless